hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -32,6 +32,69 @@ case class TableStage(
   def contextRef: IR = Ref("context", contextType)
 
   def globals: IR = GetField(broadcastRef, globalsField)
+
+  def adjustPartitionsTo(newPartitioner: RVDPartitioner): TableStage = {
+    require(newPartitioner.satisfiesAllowedOverlap(newPartitioner.kType.size - 1))
+    require(newPartitioner.kType.isPrefixOf(partitioner.kType))
+
+    val partitionMapping: IndexedSeq[Row] = newPartitioner.rangeBounds.map { i =>
+      Row(i, partitioner.queryInterval(i))
+    }
+    val partitionMappingType = TStruct(
+      "partitionBound" -> TInterval(partitioner.kType),
+      "parentPartitions" -> TArray(TInt32)
+    )
+
+    val prevContextUID = genUID()
+    val mappingUID = genUID()
+    val idxUID = genUID()
+    val newContexts = Let(
+      prevContextUID,
+      contexts,
+      ToArray(
+        StreamMap(
+          ToStream(
+            Literal(
+              TArray(partitionMappingType),
+              partitionMapping)),
```

Comment:
I don't think this works in general. The intervals in a partitioner can contain prefixes of the key type, so they won't necessarily be valid `TInterval(partitioner.kType)` values.